registry_provider_integration_test.go

```diff
@@ -0,0 +1,442 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegistryProvidersList(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	t.Run("with providers", func(t *testing.T) {
+		orgTest, orgTestCleanup := createOrganization(t, client)
+		defer orgTestCleanup()
+
+		createN := 10
+		providers := make([]*RegistryProvider, 0)
+		// these providers will be destroyed when the org is cleaned up
+		for i := 0; i < createN; i++ {
+			providerTest, _ := createPublicRegistryProvider(t, client, orgTest)
+			providers = append(providers, providerTest)
+		}
+		for i := 0; i < createN; i++ {
+			providerTest, _ := createPrivateRegistryProvider(t, client, orgTest)
+			providers = append(providers, providerTest)
+		}
+		providerN := len(providers)
+		publicProviderN := createN
+
+		t.Run("returns all providers", func(t *testing.T) {
+			returnedProviders, err := client.RegistryProviders.List(ctx, orgTest.Name, &RegistryProviderListOptions{
+				ListOptions: ListOptions{
+					PageNumber: 0,
+					PageSize:   providerN,
+				},
+			})
+			require.NoError(t, err)
+			assert.NotEmpty(t, returnedProviders.Items)
+			assert.Equal(t, providerN, returnedProviders.TotalCount)
+			assert.Equal(t, 1, returnedProviders.TotalPages)
+			for _, rp := range returnedProviders.Items {
+				foundProvider := false
+				for _, p := range providers {
+					if rp.ID == p.ID {
+						foundProvider = true
+						break
+					}
+				}
+				assert.True(t, foundProvider, "Expected to find provider %s but did not:\nexpected:\n%v\nreturned\n%v", rp.ID, providers, returnedProviders)
+			}
+		})
+
+		t.Run("returns pages", func(t *testing.T) {
```

Comment:
For this kind of test, the usual convention is to set `PageSize` and `PageNumber` to some incredibly large number, which I think is the easiest way to verify pagination is working as intended. Here's an example:

https://github.com/hashicorp/go-tfe/blob/11a4eb9078cef3c9caf389325ddc6327e749ea94/workspace_integration_test.go#L44-L58